homework7/hw7_task1/task_find_occurrences/find_occurrences.py

```diff
@@ -0,0 +1,30 @@
+"""
+Given a dictionary (tree), that can contains multiple nested structures.
+Write a function, that takes element and finds the number of occurrences
+of this element in the tree.
+Tree can only contains basic structures like:
+    str, list, tuple, dict, set, int, bool
+"""
+from typing import Any, Iterable, Iterator, Mapping
+
+
+def recursively_read(tree: Iterable, element) -> int:
+    occurrences = 0
+    iterator: Iterator
+    if isinstance(tree, Mapping):
+        iterator = iter(tree.values())
+    else:
+        iterator = iter(tree)
+    while next_element := next(iterator, None):
```

Comment:
What about `0` or an empty list?  `tree = {"empty_list": [], "zero": 0}`

There is 2 way to write a cycle in Python. can you name them?

Reply:
While and for. 
In case of my realization I use default value in `next()` to prevent StopIteration exception, but as you noticed `None` is not the best suit for handling while condition just because we have number of `Nones` like 0, [], which are also can be placed in tree. 
So, one possible solution to avoid this issue is to create **unique** object like `sentinel = object()` for managing StopIteration (mean using as default value for `next()` and then include it in while condition).